crates/core/src/generator/trait.rs

```diff
@@ -421,8 +436,32 @@ where
                         // converts a FunctionCallDefinition to a NamedTxRequest (filling in fuzzable args),
                         // returns a callback handle and the processed tx request
                         let prepare_tx = |req| {
-                            let args = get_fuzzed_args(req, &canonical_fuzz_map, i);
+                            let mut args = get_fuzzed_args(req, &canonical_fuzz_map, i);
                             let fuzz_tx_value = get_fuzzed_tx_value(req, &canonical_fuzz_map, i);
+                            // Special handling for WorldID proof generation
+                            if req
+                                .kind
+                                .as_ref()
+                                .is_some_and(|k| k == "groth16-verifyProof")
+                            {
+                                let a_point = crate::generator::bn254_points::generate_g1_point(i);
+                                let b_point = crate::generator::bn254_points::generate_g2_point(i);
+                                let c_point =
+                                    crate::generator::bn254_points::generate_g1_point(i + 1000); // Offset for uniqueness
+                                let proof_values = [
+                                    a_point[0], a_point[1], // a.x, a.y
+                                    b_point[1], b_point[0], // b.x1, b.x0
+                                    b_point[3], b_point[2], // b.y1, b.y0
+                                    c_point[0], c_point[1], // c.x, c.y
+                                ];
+                                let proof_str = proof_values
+                                    .iter()
+                                    .map(|v| format!("0x{:064x}", v))
+                                    .collect::<Vec<_>>()
+                                    .join(", ");
+                                args[5] = format!("[{}]", proof_str);
+                            }
```

Comment:
I'm a little iffy on scenario-specific logic living inside the `Generator` trait, but it works for now. While it does make sense to have this kind of logic in the generator, this closure could get really big and hard to read if we add support for any other scenarios here. I'll work out a more modular solution so we can support more scenarios in the future, but this is fine for now.